pkg/network/common/common.go

```diff
@@ -276,3 +278,30 @@ func RegisterSharedInformerEventHandlers(kubeInformers kinternalinformers.Shared
 		},
 	})
 }
+
+var (
+	ErrorNetworkInterfaceNotFound = fmt.Errorf("could not find network interface")
+)
+
+func GetLinkDetails(ip string) (netlink.Link, *net.IPNet, error) {
+	links, err := netlink.LinkList()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, link := range links {
+		addrs, err := netlink.AddrList(link, netlink.FAMILY_V4)
+		if err != nil {
+			glog.Warningf("Could not get addresses of interface %q: %v", link.Attrs().Name, err)
+			continue
+		}
+
+		for _, addr := range addrs {
+			if addr.IP.String() == ip {
+				return link, addr.IPNet, nil
```

Comment:
This is not the same as what findEgressLink() was doing. The `netlink.Addr`'s embedded `IPNet` is an IP address (eg "172.17.0.**3**/24"), but `eip.localEgressNet` is supposed to be the subnet CIDR that includes it (eg "172.17.0.**0**/24"). Hence the re-parsing of it (because there's no public method to directly mask the IP address form into the subnet form).